test/src/androidTest/kotlin/io/realm/InstrumentedTests.kt

```diff
@@ -136,25 +149,156 @@ class InstrumentedTests {
         realm.commitTransaction()
     }
 
-    // FIXME API-CLEANUP Local implementation of pointer wrapper to support test. Using the internal
-    //  one would require jni-swig-stub to be api dependency from cinterop/library. Don't know if
-    //  the test is needed at all at this level
-    //  https://github.com/realm/realm-kotlin/issues/56
-    class LongPointerWrapper(val ptr: Long) : io.realm.runtimeapi.NativePointer
     @Test
-    fun testRealmModelInternalPropertiesGenerated() {
-        val p = Sample()
-        val realmModel: RealmModelInternal = p as? RealmModelInternal ?: error("Supertype RealmModelInternal was not added to Sample class")
-
-        // Accessing getters/setters
-        realmModel.`$realm$IsManaged` = true
-        realmModel.`$realm$ObjectPointer` = LongPointerWrapper(0xCAFEBABE)
-        realmModel.`$realm$Pointer` = LongPointerWrapper(0XCAFED00D)
-        realmModel.`$realm$TableName` = "Sample"
-
-        assertEquals(true, realmModel.`$realm$IsManaged`)
-        assertEquals(0xCAFEBABE, (realmModel.`$realm$ObjectPointer` as LongPointerWrapper).ptr)
-        assertEquals(0XCAFED00D, (realmModel.`$realm$Pointer` as LongPointerWrapper).ptr)
-        assertEquals("Sample", realmModel.`$realm$TableName`)
+    fun garbageCollectorShouldFreeNativeResources() {
+        val command = arrayListOf("/system/bin/sh", "-c", "cat /proc/${Process.myPid()}/maps | grep default.realm | awk '{print \$1}'")
+
+        var realm: Realm? = openRealmFromTmpDir()
+
+        // Building a 1MB String
+        val oneMBstring = StringBuilder("").apply {
+            for (i in 1..4096) {
+                // 128 length (256 bytes)
+                append("v7TPOZtm50q8kMBoKiKRaD2JhXgjM6OUNzHojXuFXvxdtwtN9fCVIW4njdwVdZ9aChvXCtW4nzUYeYWbI6wuSspbyjvACtMtjQTtOoe12ZEPZPII6PAFTfbrQQxc3ymJ")
+            }
+        }.toString()
+        val oneMB = 1048576L
+
+        var mappedMemorySize = numberOfMemoryMappedBytes(command)
+        assertTrue(mappedMemorySize < oneMB, "Opening a Realm should not cost more than 12KB")
+
+        // inserting ~ 100MB of data and keep a strong reference to all allocated objects
+        val referenceHolder = mutableListOf<Sample>()
+        realm!!.beginTransaction()
```

Comment:
Might be that there are too fundamental differences in the operation on platforms, but it seems like much of the test code is the same on the platforms and only the memory footprint calculations are differs. Could it be commonized into one implementation with some platform specific utility methods?

Reply:
Didn't want to jump into this until we figure out how to run and filter specific platform test (like for `iOS`)